programs/psinode/main.cpp

```diff
@@ -362,14 +363,23 @@ namespace psibase
       const auto& s = boost::program_options::validators::get_single_string(values);
       if (s.starts_with('{'))
       {
-         auto key = psio::convert_from_json<ClaimKey>(s);
-         if (key.service.str() != "verifyec-sys")
+         auto                    key = psio::convert_from_json<ClaimKey>(s);
+         std::shared_ptr<Prover> result;
+
+         if (key.service.str() == "verifyec-sys")
+         {
+            result = std::make_shared<EcdsaSecp256K1Sha256Prover>(
+                key.service, psio::from_frac<PrivateKey>(key.rawData));
+         }
+         else if (key.service.str() == "verify-sys")
+         {
+            result = std::make_shared<OpenSSLProver>(key.service, key.rawData);
```

Comment:
Shouldn't the key data be fracpacked?

Reply:
Why? It's already in a binary format.

Reply:
Just noticed that the k1 key is expected to be packed, why is it different?

Reply:
The k1 key is actually a `variant<k1_key, r1_key>`, so it needs additional encoding. The BER encoded key already contains a discriminator to identify the type of key. I don't see any reason to add another layer of wrapping.